core/jobs/types/firebase_adapters.py

```
@@ -0,0 +1,109 @@
+
+        Raises:
+            TypeError. Always, since Firebase records lack a valid `user_id`.
+        """
+        raise TypeError('Firebase records have no Oppia User ID')
+
+    @classmethod
+    def from_oppia_models(
+        cls,
+        settings: user_models.UserSettingsModel,
+        auth_details: auth_models.UserAuthDetailsModel,
+    ) -> 'WeakRecord | None':
```

Ruff (0.15.15)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)